src/main/java/com/palantir/gradle/revapi/RevapiPlugin.java

```diff
@@ -60,14 +61,40 @@ public void apply(Project project) {
                     Configuration revapiNewApi = project.getConfigurations().create("revapiNewApi", conf -> {
                         conf.extendsFrom(
                                 project.getConfigurations().getByName(JavaPlugin.COMPILE_CLASSPATH_CONFIGURATION_NAME));
+                        conf.setCanBeConsumed(false);
                     });
 
-                    task.dependsOn(allJarTasksIncludingDependencies(project, revapiNewApi));
+                    Configuration revapiNewApiElements = project.getConfigurations()
+                            .create("revapiNewApiElements", conf -> {
+                                conf.extendsFrom(project.getConfigurations()
+                                        .getByName(JavaPlugin.API_ELEMENTS_CONFIGURATION_NAME));
+                                conf.setCanBeConsumed(false);
+                            });
+
+                    Provider<Set<Jar>> allJarTasks = allJarTasksIncludingDependencies(project, revapiNewApi);
+
+                    task.dependsOn(allJarTasks);
                     task.getAcceptedBreaks().set(acceptedBreaks(project, configManager, extension.oldGroupAndName()));
 
-                    Jar jarTask = project.getTasks().withType(Jar.class).getByName(JavaPlugin.JAR_TASK_NAME);
-                    task.getNewApiJars().set(jarTask.getOutputs().getFiles());
-                    task.getNewApiDependencyJars().set(revapiNewApi);
+                    task.getNewApiJars().set(GradleUtils.memoisedProvider(project, () -> {
+                        File thisJarFile = project.getTasks()
+                                .withType(Jar.class)
+                                .getByName(JavaPlugin.JAR_TASK_NAME)
+                                .getOutputs()
+                                .getFiles()
+                                .getSingleFile();
+
+                        Set<File> allJarFiles = allJarTasks.get().stream()
+                                .flatMap(jar -> jar.getOutputs().getFiles().getFiles().stream())
+                                .collect(Collectors.toSet());
+
+                        return Sets.union(
+                                Collections.singleton(thisJarFile),
+                                Sets.intersection(revapiNewApiElements.resolve(), allJarFiles));
```

Comment:
here you essentially want "external dependencies", so excluding projects.
You can do that much nicer directly from the configuration like so, and it gives you a lazy `FileCollection` too:

```java
conf.getIncoming()
        .artifactView(vc -> vc.componentFilter(ci -> ci instanceof ModuleComponentIdentifier))
        .getFiles()
```

Then you don't need the intersection, that is the only thing FileCollections don't support